lib/parallel_tests/cli.rb

```diff
@@ -255,6 +261,10 @@ def parse_options!(argv)
         raise "--group-by #{allowed.join(" or ")} is required for --only-group"
       end
 
+      if options[:isolate_count] && options[:isolate_count] >= ParallelTests.determine_number_of_processes(options[:count])
+        abort 'Number of isolated processes must be less than total the number of processes'
+      end
```

Comment:
this triggers determine_number_of_processes twice when count was not given and duplicates the `count ||= determine_number_of_processes` logic
so this check should move to where count is set or count setting should move here

Reply:
I tried moving num_processes determination to parse_options. Lot of specs fail - as they don't expect `:count` option. That does not feel right either - as parse_option should be just that.

I prefer validation bounds check for all options inside parse_options. This keeps the code in appropriate places at the cost of calling determine_number_of_processes twice.

But I did move the validation for this inside the main `run` method

Reply:
Well, that did not work as the specs fail as CLI specs call only the parse_options method.

I would leave it up to you.. Three ways to fix this...

1. leave it as is.
2. Ignore the bounds check.
3. Do not fail on isolate_count being out of bounds. Instead coerce the isolate_count to  (n-1) in the in_even_groups_by_size by size method.

Reply:
@grosser ☝️  In case you have not seen the above reply yet.

Reply:
4. fail with a nice error when doing the group logic ? :)

Reply:
Done... Please do check the whole PR again with fresh set of eyes.

Reply:
It is failing like this:

![image](https://user-images.githubusercontent.com/122798/93376269-57e10880-f827-11ea-9c4e-12e0f3a0b1f5.png)

or

![image](https://user-images.githubusercontent.com/122798/93376460-9e366780-f827-11ea-8ca4-aaaaf9efcf9e.png)

Specs pass locally.

Any ideas?